src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -177,6 +159,27 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
             }
         }
 
+        private static void ABTest(Document document, Workspace workspace)
+        {
+            // AB Test
+            // If the feature is OFF and the feature options have not yet been set to their assigned flight, do so now
+            if (!workspace.Options.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting, document.Project.Language)
+                && !workspace.Options.GetOption(CodeCleanupABTestOptions.ParticipatedInExperiment))
+            {
+                var experimentationService = document.Project.Solution.Workspace.Services.GetService<IExperimentationService>();
+
+                if (experimentationService != null)
+                {
+                    if (experimentationService.IsExperimentEnabled(s_experimentName))
+                    {
+                        workspace.Options = workspace.Options.WithChangedOption(new OptionKey(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting, document.Project.Language), true);
```

Comment:
won't this permanently opt the user into having this behavior?  What happens if we don't like the results of the experiment?  How do we get the people that were opted-in, back to the old behavior?

Reply:
Users can manually go to tools->options to uncheck the PerformAdditionalCodeCleanupDuringFormatting checkbox top to opt out. 
![image](https://user-images.githubusercontent.com/17056651/43162639-84b1bcf8-8f40-11e8-84a0-7fc4e6cb1ec4.png)

@kuhlenh Are we ok with this?

Reply:
That isn't how an A/B experiment is supposed to work. :)   You use the AB test to run an experiement and to get statistical data.   Once you have that data you can make a determination about what direction to actually go.  And that's what should be done to everyone (until the next experiment comes along).  Leaving some portion of your users just stuck in their experiment and forcing htem to opt-out of it isn't appropriate.